src/SassBuilder.php

```diff
@@ -90,13 +90,33 @@ public function setOutput(SymfonyStyle $output): void
     /**
      * @internal
      */
-    public static function guessCssNameFromSassFile(string $sassFile, string $outputDirectory): string
+    public static function guessCssNameFromSassFile(string $sassFile, string $outputDirectory, string|int $fileName = null): string
     {
-        $fileName = basename($sassFile, '.scss');
+        if (null === $fileName || \is_int($fileName)) {
+            $fileName = basename($sassFile, '.scss');
+        }
 
         return $outputDirectory.'/'.$fileName.'.output.css';
     }
 
+    public function getIdentifierByLogicalPath(string $path): ?string
+    {
+        if (array_is_list($this->sassPaths)) {
+            return null;
+        }
+
+        foreach ($this->sassPaths as $identifier => $configuredSassPath) {
+            // as the configured paths include the project dir, we need to subtract it to be able to compare the paths
+            $logicalPath = str_replace($this->projectRootDir.'/assets/', '', $configuredSassPath);
```

Comment:
be careful about windows paths when doing `str_replace`.

And also be careful that you don't remove only a prefix. If the project root dir is `/app`, things will be if you have a path like `/app/assets/sub/app/assets`

Reply:
Good one, I refactored it to make use of `substr`. How would I need to address any Windows related problems?